addons/desec.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+# T&M Hansson IT AB © - 2020, https://www.hanssonit.se/
+
+true
+SCRIPT_NAME="deSEC Registration"
+SCRIPT_EXPLAINER="This script will automatically register a domain of your liking, and set it to automatically update your external IP address."
+# shellcheck source=lib.sh
+source /var/scripts/fetch_lib.sh || source <(curl -sL https://raw.githubusercontent.com/nextcloud/vm/master/lib.sh)
+
+# Maybe move to lib?
+DEDYNPASS=$(gen_passwd "$SHUF" "a-zA-Z0-9")
+
+# Enter the subdomain
+msg_box "Please enter the subdomain (*example*.dedyn.io) that you want to use"
+while :
+do
+    SUBDEDYN=$(input_box_flow "Please enter the subdomain (*example*.dedyn.io) that you want to use \
+The only allowed characters for the username are:
+'a-z', 'A-Z', and '0-9'")
+    if [[ "$SUBDEDYN" == *" "* ]]
+    then
+        msg_box "Please don't use spaces."
+    elif [ "${SUBDEDYN//[A-Za-z0-9]}" ]
+    then
+        msg_box "Allowed characters for the username are:\na-z', 'A-Z', and '0-9'\n\nPlease try again."
+    else
+        DEDYNDOMAIN="$SUBDEDYN.dedyn.io"
+        break
+    fi
```

Comment:
we should already check in this loop if the subdomain is avaliable. Is there an API for that?

Reply:
You can query the DNS to see if the subdomain has an SOA record. If yes, it is taken. If not, it generally is available (with the exception of some reserved names, in particular names like `autoconfig` and `autodiscover`, and names beginning with `_`).